src/libs/ajax/data-table-providers/WdsDataTableProvider.ts

```diff
@@ -82,33 +84,61 @@ const getRelationParts = (val: unknown): string[] => {
   return []
 }
 
-// Extract wds URL from Leo response. exported for testing
-export const getWdsUrl = apps => {
-  // look explicitly for an app named 'wds-${app.workspaceId}'. If found, use it, even if it isn't running
-  // this handles the case where the user has explicitly shut down the app
+// Invokes logic to determine a URL for WDS
+// If WDS is not running, a URL will not be present -- in some cases, this function may invoke
+// a new call to Leo to instantiate a WDS being available, thus having a valid URL
+export const resolveWdsUrl = (apps, workspaceId) => {
+  // WDS looks for Kubernetes deployment statuses (such as RUNNING or PROVISIONING), expressed by Leo
+  // See here for specific enumerations -- https://github.com/DataBiosphere/leonardo/blob/develop/core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala
+  // look explicitly for a RUNNING app named 'wds-${app.workspaceId}' -- if WDS is healthy and running, there should only be one app RUNNNING
   const namedApp = apps.filter(app => app.appType === 'CROMWELL' && app.appName === `wds-${app.workspaceId}` && app.status === 'RUNNING')
   if (namedApp.length === 1) {
     return namedApp[0].proxyUrls.wds
   }
-  // if we didn't find the expected app 'cbas-wds-default', go hunting:
-  const candidates = apps.filter(app => app.appType === 'CROMWELL' && app.status === 'RUNNING')
+
+  // if we didn't find the expected app 'wds-${app.workspaceId}' running...
+  const candidates = apps.filter(app => app.appType === 'CROMWELL' && app.appName === `wds-${app.workspaceId}`)
+  // ...nothing has launched yet, bring WDS to life!
   if (candidates.length === 0) {
-    // no app deployed yet
+    // !getConfig().isProd is temporary until WDS is ready for production launch in Feb 2023
+    if (!getConfig().isProd) {
+      withErrorReporting('An error occurred when creating your data tables. Please reach out to support@terra.bio', async () => {
+        await Ajax().Apps.createAppV2(`wds-${workspaceId}`, workspaceId)
+      })
+    }
+    return ''
+  }
```

Comment:
nit: you can leave out the `if (!getConfig().isProd)`. Since the Data tab is hidden, users can't get to it and therefore have no way of triggering this code. In other words, the `if (!getConfig().isProd)` check is at the Data tab level so we don't need it here.

Reply:
Removed!